core/src/ast/types.rs

```diff
@@ -372,102 +372,45 @@ impl TypeName {
     pub fn to_syn(&self) -> syn::Type {
         match self {
             TypeName::Primitive(name) => {
-                syn::Type::Path(syn::parse_str(PRIMITIVE_TO_STRING.get(name).unwrap()).unwrap())
+                let s = PRIMITIVE_TO_STRING.get(name).unwrap();
+                let ident = proc_macro2::Ident::new(s, Span::call_site());
```

Comment:
`Ident`s are not `Sync`.

Reply:
(anything that includes `Span` gets pretty sketchy pretty quickly when it comes to serialization/threading, one of the benefits of Diplomat's AST is that it doesn't have that problem)

Reply:
`PRIMITIVE_TO_IDENT` doesn't need to be a static map, it can be a function. Matching over a few strings isn't expensive, and in this case it'd be cleaner not to go through strings.